# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libANGLE/Constants.h
# Commit: 3a91885d47e0
# Full Hash: 3a91885d47e0ef7fba9749562d52300eea67c651
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-11-11 21:52:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 77cee77c0307 -r 3a91885d47e0 gfx/angle/checkout/src/libANGLE/Constants.h
--- a/gfx/angle/checkout/src/libANGLE/Constants.h	Fri Nov 08 16:41:42 2019 +0000
+++ b/gfx/angle/checkout/src/libANGLE/Constants.h	Fri Nov 08 17:06:00 2019 +0000
@@ -11,6 +11,8 @@
 
 #include "common/platform.h"
 
+#include <stdint.h>
+
 namespace gl
 {
 
@@ -41,7 +43,11 @@
     // GL_EXT_geometry_shader increases the minimum value of GL_MAX_UNIFORM_BUFFER_BINDINGS to 48.
     IMPLEMENTATION_MAX_UNIFORM_BUFFER_BINDINGS = 48,
 
-    IMPLEMENTATION_MAX_TRANSFORM_FEEDBACK_BUFFERS = 4,
+    // Transform feedback limits set to the minimum required by the spec.
+    IMPLEMENTATION_MAX_TRANSFORM_FEEDBACK_INTERLEAVED_COMPONENTS = 64,
+    IMPLEMENTATION_MAX_TRANSFORM_FEEDBACK_SEPARATE_ATTRIBS       = 4,
+    IMPLEMENTATION_MAX_TRANSFORM_FEEDBACK_SEPARATE_COMPONENTS    = 4,
+    IMPLEMENTATION_MAX_TRANSFORM_FEEDBACK_BUFFERS                = 4,
 
     // Maximum number of views which are supported by the implementation of ANGLE_multiview.
     IMPLEMENTATION_ANGLE_MULTIVIEW_MAX_VIEWS = 4,
@@ -64,7 +70,18 @@
 
     // Maximum number of slots allocated for atomic counter buffers.
     IMPLEMENTATION_MAX_ATOMIC_COUNTER_BUFFERS = 8,
+
+    // Implementation upper limits, real maximums depend on the hardware.
+    IMPLEMENTATION_MAX_SHADER_STORAGE_BUFFER_BINDINGS = 64
 };
+
+namespace limits
+{
+// Some of the minimums required by GL, used to detect if the backend meets the minimum requirement.
+// Currently, there's no need to separate these values per spec version.
+constexpr uint32_t kMinimumComputeStorageBuffers = 4;
+}  // namespace limits
+
 }  // namespace gl
 
 #endif  // LIBANGLE_CONSTANTS_H_